core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -267,6 +267,10 @@ private[spark] class ExecutorAllocationManager(
     (numRunningOrPendingTasks + tasksPerExecutor - 1) / tasksPerExecutor
   }
 
+  private def totalRunningTasks(): Int = synchronized {
```

Comment:
Looks like no one invoke this method?

Reply:
This is being called from the test.

Reply:
I'm not sure why do we need to add a method which only used for unit test. If want to verify the behavior of `totalRunningTasks`, I think `maxNumExecutorsNeeded` can also be used indirectly for verification.

Reply:
Its okay to add a method which is used for unit testing purpose only. I am not inclined towards the idea of using `maxNumExecutorsNeeded` to indirectly verify `totalRunningTasks` for the following reason - 

Currently, the test case is testing what it is supposed to. If you check for `maxNumExecutorsNeeded` instead, it might not be clear what we are testing.